ehitchhiking-ui/src/app/shared/modals/profile-modal/car-info-form/car-info-form.component.ts

```diff
@@ -0,0 +1,55 @@
+import {Component, Input, OnInit, Output, EventEmitter} from '@angular/core';
+import {FormGroup, FormBuilder, Validators} from '@angular/forms';
+import {Car} from '../../../models/car';
+@Component({
+	selector: 'app-car-info-form',
+	templateUrl: './car-info-form.component.html',
+	styleUrls: ['./car-info-form.component.sass'],
+})
+export class CarInfoFormComponent implements OnInit {
+	@Input() carInfoForm: FormGroup;
+	@Input() addCarMod: boolean;
+	@Input() carIndex: number;
+	@Output() onChangeCarInfo = new EventEmitter<Car>();
+	addCarForm: FormGroup;
+
+	constructor(private formBuilder: FormBuilder) {}
+
+	ngOnInit() {
+		if (this.addCarMod) {
+			this.initForm();
+		}
+	}
+
+	initForm(): void {
+		this.addCarForm = this.formBuilder.group({
+			model: ['', [Validators.required, Validators.pattern('^[0-9a-zA-Z-]{3,50}$')]],
+			color: ['', [Validators.required, Validators.pattern('^[a-zA-Z-]{3,20}$')]],
+			carNumber: ['', [Validators.required, Validators.pattern('^[0-9a-zA-Z-]{6,8}$')]],
+			experience: ['', [Validators.required, Validators.pattern('^[0-9]{1,2}$'), Validators.max(80)]],
+		});
+	}
+
+	public hasError(controlName: string, errorName: string) {
```

Comment:
In the TypeScript public is default access modifier, hence there is no reason to write public

Reply:
done